### DISCUSSION TITLE
typespec-poc with APIM

### DISCUSSION THREAD

**Discussion by Xingo123** [microsoft/typespec#2905] [Ideas]
Hi Guys,

I've created 10 months ago a repo with Github actions and TypeSpec to do the API-First way of working. I would like to have some feedback on it also? Can you guys help a geek out? :D 

https://github.com/Xingo123/typespec-poc 

Xing Woo

**Comment by timotheeguerin** [comment]
Hi @Xingo123 I think this is a great way of doing so, the one thing I would maybe consider as a flow is actually commiting the generated openapi spec to the repo and instead having a CI step to make sure it is up to date. Reason being you are now providing your openapi spec directly and makes it easier for anybody else that needs to consume it.

**Reply by Xingo123** [reply]
Hi @timotheeguerin,

 I agree with the feedback! From a API-First provider perspective you want to make sure that the openapi-spec in code is your final deliverable.

Thanks for the feedback.

Other question, How well is TypeSpec adopted within Enterprises? Because Enterprises thinks they understand API-First principle, but still the develop endpoints and let it generate open api spec?

I would like to have the lessons learned, because I want to try to implement at my current client.

Kind regards,

Xing Woo

**Reply by bterlson** [reply]
We have use in a few enterprises (Microsoft being a notable example 😁) but it's still early days for us. I agree that code-first approaches that treat API specs as a build artifact are quite common, and we hope that TypeSpec will offer enough benefit that more folks will clearly see teh value of an API first approach.

**Reply by Xingo123** [reply]
Hi @bterlson, coming back to adopting within Enterprises. At my client we will have a hackathon:  "API Management: Developing and managing APIs for better service integration and innovation". Colleagues of my already have submitted a "where you can publish a TypeSpec definition and download various derived assets".

So I would also like to do a TypeSpec i.c.m. with API Management and Kiota which I already did in my fork https://github.com/Xingo123/typespec-poc. But thinking I would like to know if you guys have some ideas regarding the subject, which you don't have time to execute the idea. I think it's a good way to inner source the concept when we have Hackathon it.
